nni/algorithms/compression/pytorch/auto_compress/experiment.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+
+import dataclasses
+from pathlib import Path
+from subprocess import Popen
+from typing import Optional
+
+import nni
+from nni.experiment import Experiment, ExperimentConfig, AlgorithmConfig
+
+
+class AutoCompressExperimentConfig(ExperimentConfig):
+    auto_compress_module_file_name: str = './auto_compress_module.py'
+
+    def __setattr__(self, key, value):
+        fixed_attrs = {'trial_command': 'python3 -m nni.algorithms.compression.pytorch.auto_compress.trial_entry'}
+        if key in fixed_attrs and type(value) is not type(dataclasses.MISSING) and not value.startswith(fixed_attrs[key]):
+            raise AttributeError(f'{key} is not supposed to be set in AutoCompress mode by users!')
+        # 'trial_code_directory' is handled differently because the path will be converted to absolute path by us
+        if key == 'trial_code_directory' and not (value == Path('.') or os.path.isabs(value)):
+            raise AttributeError(f'{key} is not supposed to be set in AutoCompress mode by users!')
```

Comment:
this is not a correct error message...

Reply:
and why the value must be Path('.') or absolute path?

Reply:
have removed this.